### PR TITLE
Remove override methods for ESGatewayDeployment 

### PR DESCRIPTION
**Pull request by nelljerram**
Since https://github.com/tigera/operator/pull/3613, we no longer need these methods for each customization structure that we define.

## For PR author

- [ ] Tests for change.
- [ ] If changing pkg/apis/, run `make gen-files`
- [ ] If changing versions, run `make gen-versions`

## For PR reviewers

A note for code reviewers - all pull requests must have the following:

- [ ] Milestone set according to targeted release.
- [ ] Appropriate labels:
  - `kind/bug` if this is a bugfix.
  - `kind/enhancement` if this is a a new feature.
  - `enterprise` if this PR applies to Calico Enterprise only.

### CODE REVIEW

**Comment by nelljerram** [comment]
@vara2504 I've requested your review on this, but you may like to wait until #3626 is merged first, because then the delta here will be smaller.

**Review by caseydavenport — APPROVED** [review]

**Pull request merged**
